### PR TITLE
[ADD] fiscal localizations: managing negative discount lines

### PR DESCRIPTION
**Pull request by vrpa-odoo**
Clarifying the application of managing negative lines, this commit elucidates how to effectively utilize the global discount setting. The inclusion of this feature enhances user understanding and proficiency, providing insights into the functionality of the newly introduced setting for handling negative discount lines.

task: 3638605

### CODE REVIEW

**Review by chiaraprattico — CHANGES_REQUESTED** [review]
Hi @vrpa-odoo nice one, I only added a few comments. This only addresses master and no earlier versions?

**Comment by vrpa-odoo** [comment]
> Hi @vrpa-odoo nice one, I only added a few comments. This only addresses master and no earlier versions?

Yes, the setting is specific to the master. One can perform this functionality in earlier versions by adding it to System Parameters.

File: content/applications/finance/fiscal_localizations/india.rst [hunk 1]
```diff
@@ -166,6 +166,56 @@ Cancel`.
 
 .. _india/verify-e-invoice:
 
+Managing Negative Lines in E-Invoice
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Government portal prohibits the submission of data with negative line values. To facilitate the inclusion
+of discount lines, users are required to activate the option for managing negative lines.
+This ensures seamless handling of discounts within the e-invoicing process.
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 2]
```diff
@@ -166,6 +166,56 @@ Cancel`.
...
+
+To enable the this service in Odoo
+
+Go to :menuselection:`Accounting --> Configuration --> Settings
+--> Manage Negative Lines` and turn on :guilabel:`Manage Negative Lines`.
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 3]
```diff
@@ -166,6 +166,56 @@ Cancel`.
...
+
+.. image:: india/manage-negative-lines.png
+   :alt: Managing the negative discount lines
+
+
+**Negative Lines Conversion in E-Invoice**
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 4]
```diff
@@ -166,6 +166,56 @@ Cancel`.
...
+
+In Odoo, when dealing with E-invoice entries, there may be scenarios where negative lines are encountered. These negative lines are typically employed to represent discounts or adjustments associated with specific products or global discount.
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 5]
```diff
@@ -166,6 +166,56 @@ Cancel`.
...
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Product A             | 123456       | 1,000            | 1            |    18%       |    1,180  |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Product B             | 239345       | 1,500            | 2            |    5%        |    3,150  |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Discount on Product A | 123456       | -100             | 1            |    18%       |    -118   |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+
+
+To streamline and enhance clarity, negative lines are converted into positive discounts based on the HSN Code and GST rate. Here's the transformed representation:
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 6]
```diff
@@ -166,6 +166,55 @@ Cancel`.
 
 .. _india/verify-e-invoice:
 
+Managing Negative Lines in E-Invoice
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The government portal prohibits the submission of data with negative line values.
+To facilitate the inclusion of discount lines, activating the option to manage negative lines is necessary.
+
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/india.rst:174]: Need to remove one space

File: content/applications/finance/fiscal_localizations/india.rst [hunk 7]
```diff
@@ -166,6 +166,55 @@ Cancel`.
...
+
+To enable it go to :menuselection:`Accounting --> Configuration --> Settings
+--> Manage Negative Lines` and turn on :guilabel:`Manage Negative Lines`.
+
+.. image:: india/manage-negative-lines.png
+   :alt: Managing the negative discount lines
+
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/india.rst:181]: Need to remove one space

File: content/applications/finance/fiscal_localizations/india.rst [hunk 8]
```diff
@@ -166,6 +166,55 @@ Cancel`.
...
+| **Product Name**      | **HSN Code** | **Tax Excluded** | **Quantity** | **GST Rate** | **Total** |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Product A             | 123456       | 1,000            | 1            |    18%       |    1,180  |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Product B             | 239345       | 1,500            | 2            |    5%        |    3,150  |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Discount on Product A | 123456       | -100             | 1            |    18%       |    -118   |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+
+
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/india.rst:202]: Need to remove one space

File: content/applications/finance/fiscal_localizations/india.rst [hunk 9]
```diff
@@ -166,6 +166,55 @@ Cancel`.
...
+| **Product Name** | **HSN Code** | **Tax Excluded** | **Quantity** | **Discount** | **GST Rate** | **Total** |
++------------------+--------------+------------------+--------------+--------------+--------------+-----------+
+| Product A        | 123456       | 1,000            | 1            | 100          |  18%         |  1,062    |
++------------------+--------------+------------------+--------------+--------------+--------------+-----------+
+| Product B        | 239345       | 1,500            | 2            | 0            |  5%          | 3,150     |
++------------------+--------------+------------------+--------------+--------------+--------------+-----------+
+
+In this conversion, negative lines have been transformed into positive discounts, maintaining accurate calculations based on the HSN Code and GST rate. This ensures a more straightforward and standardized representation in the E-invoice records.
+
+
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/india.rst:217]: Need to remove one space

**Review by chiaraprattico — COMMENTED** [review]
@vrpa-odoo pinging the final review team to validate but there are some extra lines that need.

File: content/applications/finance/fiscal_localizations/india.rst (hunk 6 above)
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst:174]: done

File: content/applications/finance/fiscal_localizations/india.rst (hunk 7 above)
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst:180]: done

File: content/applications/finance/fiscal_localizations/india.rst (hunk 8 above)
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst (hunk 9 above)
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 10]
```diff
@@ -166,6 +166,45 @@ Cancel`.
...
++==================+==============+==================+==============+==============+==============+===========+
+| **Product Name** | **HSN Code** | **Tax Excluded** | **Quantity** | **Discount** | **GST Rate** | **Total** |
++------------------+--------------+------------------+--------------+--------------+--------------+-----------+
+| Product A        | 123456       | 1,000            | 1            | 100          |  18%         |  1,062    |
++------------------+--------------+------------------+--------------+--------------+--------------+-----------+
+| Product B        | 239345       | 1,500            | 2            | 0            |  5%          | 3,150     |
++------------------+--------------+------------------+--------------+--------------+--------------+-----------+
+
+In this conversion, negative lines have been transformed into positive discounts, maintaining accurate calculations based on the HSN Code and GST rate. This ensures a more straightforward and standardized representation in the E-invoice records.
+
```
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/india.rst:207]: If I understand correctly, this explanation is not specific to the example above so I suggest moving it to the intro (after "which means they need to be converted based on the HSN code and GST rate")

also : e-Invoice records

File: content/applications/finance/fiscal_localizations/india.rst [hunk 11]
```diff
@@ -166,6 +166,45 @@ Cancel`.
...
+The government portal prohibits the submission of data with negative line values.
+To facilitate the inclusion of discount lines, effective handling of negative entries becomes essential.
+
+Negative Lines Conversion in E-Invoice
+**************************************
+
+When dealing with E-invoice entries, there may be scenarios where negative lines are encountered. These negative lines are typically employed to represent discounts or adjustments associated with specific products HSN and tax rate or global discounts.
+
+Consider the following example:
```
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/india.rst:180]: I would put the example in a an "example" box

**Review by auva-odoo — CHANGES_REQUESTED** [review]
Hi @vrpa-odoo thanks for your work! I suggested a few changes to clarify the context and simplify the structure. Please also keep in mind:
- the 100 character limit per line;
- remove any previously added images if you're not using them anymore.
Thank you!

File: content/applications/finance/fiscal_localizations/india.rst [hunk 12]
```diff
@@ -166,6 +166,45 @@ Cancel`.
 
 .. _india/verify-e-invoice:
 
+Managing Negative Lines in E-Invoice
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 13]
```diff
@@ -166,6 +166,45 @@ Cancel`.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The government portal prohibits the submission of data with negative line values.
+To facilitate the inclusion of discount lines, effective handling of negative entries becomes essential.
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 14]
```diff
@@ -166,6 +166,45 @@ Cancel`.
...
+
+Negative Lines Conversion in E-Invoice
+**************************************
+
+When dealing with E-invoice entries, there may be scenarios where negative lines are encountered. These negative lines are typically employed to represent discounts or adjustments associated with specific products HSN and tax rate or global discounts.
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst [hunk 15]
```diff
@@ -166,6 +166,45 @@ Cancel`.
...
+| **Product Name**      | **HSN Code** | **Tax Excluded** | **Quantity** | **GST Rate** | **Total** |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Product A             | 123456       | 1,000            | 1            |    18%       |    1,180  |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Product B             | 239345       | 1,500            | 2            |    5%        |    3,150  |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+| Discount on Product A | 123456       | -100             | 1            |    18%       |    -118   |
++-----------------------+--------------+------------------+--------------+--------------+-----------+
+
+For clarity, negative lines are converted into positive discounts based on the HSN Code and GST rate. Here's the transformed representation:
```
[the remark this answers is not in the document]
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

File: content/applications/finance/fiscal_localizations/india.rst (hunk 10 above)
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: It is related to the above example on how we manage the negative lines.

File: content/applications/finance/fiscal_localizations/india.rst (hunk 11 above)
Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: done

**Comment by auva-odoo** [comment]
Hi @vrpa-odoo, thank you for performing the changes :) However, I noticed that there is no setting related to this anymore, so does that mean that the conversion is done automatically without users having to do anything?

**Comment by vrpa-odoo** [comment]
> Hi @vrpa-odoo, thank you for performing the changes :) However, I noticed that there is no setting related to this anymore, so does that mean that the conversion is done automatically without users having to do anything?

Yes, the conversion will be done automatically as we have added the functionality to manage the negative lines whenever the negative line is invoked

**Comment by auva-odoo** [comment]
@vrpa-odoo In this case, I would not dedicate section to it in the documentation. What do you think about adding a note somewhere (maybe in the e-Invoice system section, under "Journal"? (for reference: https://www.odoo.com/documentation/master/applications/finance/fiscal_localizations/india.html#journals), someting similar to this maybe:

Negative lines are typically used to represent discounts or adjustments associated with specific products or global discounts. Since the government portal prohibits the submission of data with negative lines, these are automatically converted into positive ones based on the product’s HSN code and GST rate.

**Comment by jva-odoo** [comment]
@auva-odoo we need an example, and I think it's not looking good if we add a big example in a note.
what do you think?

**Comment by auva-odoo** [comment]
@vrpa-odoo @jva-odoo Okay, then let's leave the section where it is now, then:
- Change the section's title to "Management of negative lines in e-Invoice" so that the title no longer implies the user needs to do actively do something. Also, could we say "e-Invoices" here? I feel it would be more gramatically correct.
- At the end of the intro, mention that the conversion is done automatically, e.g.: "This is done automatically by Odoo", or something like that.
What do you think?

File: content/applications/finance/fiscal_localizations/india.rst [hunk 16]
```diff
@@ -166,6 +166,46 @@ Cancel`.
 
 .. _india/verify-e-invoice:
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/india.rst:167]: please move this custom anchor before the "GST e-Invoice verification" heading and add this one here instead:

```suggestion
.. _india/e-invoice-negative-lines:
```

Thanks!

Comment (vrpa-odoo) [reply content/applications/finance/fiscal_localizations/india.rst]: Done

**Comment by auva-odoo** [comment]
@vrpa-odoo Thank you for your work! 
@robodoo delegate+

**Review by auva-odoo — APPROVED** [review]

**Comment by vrpa-odoo** [comment]
@robodoo r+

**Pull request closed without merging**